ormar/models/modelproxy.py

```diff
@@ -71,9 +71,14 @@ def substitute_models_with_pks(cls, model_dict: Dict) -> Dict:  # noqa  CCR001
                         )
                     model_dict[field] = pk_value
                 elif field_value:  # nested dict
-                    model_dict[field] = field_value.get(target_pkname)
+                    if isinstance(field_value, list):
+                        model_dict[field] = [
+                            target.get(target_pkname) for target in field_value
+                        ]
+                    else:
+                        model_dict[field] = field_value.get(target_pkname)
                 else:
-                    model_dict.pop(field, None)
+                    model_dict.pop(field)
```

Comment:
Please leave the `pop(field, None)` -> the whole idea here is to remove the field from dict if it's empty in data. So if the field is missing/not there for whatever reason it's ok - we don't have to remove it, but if the field is missing popping it will cause exception.